File: dmn-test-cases/standard/tck/1.1/cl2/0008-LX-arithmetic/translator/expected/java/test/_0008LXArithmeticTest.java

import java.util.*;
import java.util.stream.Collectors;

@javax.annotation.Generated(value = {"junit.ftl", "0008-LX-arithmetic.dmn"})
public class _0008LXArithmeticTest extends com.gs.dmn.runtime.JavaTimeDMNBaseDecision<Object> {
    @org.junit.jupiter.api.Test
    public void testCase001_1() {
        // Initialize inputs
        type.TLoan loan = new type.TLoanImpl(number("600000"), number("0.0375"), number("360"));

        // Initialize input context
        com.gs.dmn.runtime.Context input_ = new com.gs.dmn.runtime.Context();
        input_.add("loan", loan);

        // Check 'payment'
        com.gs.dmn.runtime.ExecutionContext context_ = com.gs.dmn.runtime.ExecutionContextBuilder.executionContext().build();
        checkValues(number("2778.69354943277"), new Payment().applyContext(input_, context_));
    }

    @org.junit.jupiter.api.Test
    public void testCase002_1() {
        // Initialize inputs
        type.TLoan loan = new type.TLoanImpl(number("30000"), number("0.0475"), number("60"));

        // Initialize input context
        com.gs.dmn.runtime.Context input_ = new com.gs.dmn.runtime.Context();
        input_.add("loan", loan);

        // Check 'payment'
        com.gs.dmn.runtime.ExecutionContext context_ = com.gs.dmn.runtime.ExecutionContextBuilder.executionContext().build();
        checkValues(number("562.707359373292"), new Payment().applyContext(input_, context_));
    }

    @org.junit.jupiter.api.Test
    public void testCase003_1() {
        // Initialize inputs
        type.TLoan loan = new type.TLoanImpl(number("600000"), number("0.0399"), number("360"));

        // Initialize input context
        com.gs.dmn.runtime.Context input_ = new com.gs.dmn.runtime.Context();
        input_.add("loan", loan);

        // Check 'payment'
        com.gs.dmn.runtime.ExecutionContext context_ = com.gs.dmn.runtime.ExecutionContextBuilder.executionContext().build();
        checkValues(number("2861.03377700389"), new Payment().applyContext(input_, context_));
    }

    private void checkValues(Object expected, Object actual) {
        com.gs.dmn.runtime.Assert.assertEquals(expected, actual);
    }
}

File: dmn-test-cases/standard/tck/1.1/cl2/0008-LX-arithmetic/translator/expected/kotlin/test/_0008LXArithmeticTest.kt

import java.util.*
import java.util.stream.Collectors

@javax.annotation.Generated(value = ["junit.ftl", "0008-LX-arithmetic.dmn"])
class _0008LXArithmeticTest : com.gs.dmn.runtime.JavaTimeDMNBaseDecision<Object>() {
    @org.junit.jupiter.api.Test
    fun testCase001_1() {
        val context_ = com.gs.dmn.runtime.ExecutionContextBuilder.executionContext().build()
        val cache_ = context_.getCache()
        // Initialize arguments
        val loan: type.TLoan? = type.TLoanImpl(number("600000"), number("0.0375"), number("360"))

        // Check 'payment'
        checkValues(number("2778.69354943277"), Payment().apply(loan, context_))
    }

    @org.junit.jupiter.api.Test
    fun testCase002_1() {
        val context_ = com.gs.dmn.runtime.ExecutionContextBuilder.executionContext().build()
        val cache_ = context_.getCache()
        // Initialize arguments
        val loan: type.TLoan? = type.TLoanImpl(number("30000"), number("0.0475"), number("60"))

        // Check 'payment'
        checkValues(number("562.707359373292"), Payment().apply(loan, context_))
    }

    @org.junit.jupiter.api.Test
    fun testCase003_1() {
        val context_ = com.gs.dmn.runtime.ExecutionContextBuilder.executionContext().build()
        val cache_ = context_.getCache()
        // Initialize arguments
        val loan: type.TLoan? = type.TLoanImpl(number("600000"), number("0.0399"), number("360"))

        // Check 'payment'
        checkValues(number("2861.03377700389"), Payment().apply(loan, context_))
    }

    private fun checkValues(expected: Any?, actual: Any?) {
        com.gs.dmn.runtime.Assert.assertEquals(expected, actual)
    }
}
